src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/utils/validation.ts

```diff
@@ -517,19 +584,28 @@ export function mapDataElementValidationToRedux(validations: IValidationIssue[],
     // for each validation, map to correct component and field key
     const componentValidations: IComponentValidations = {};
     let component;
+    let componentId;
     if (layout) {
       component = layout.find((layoutElement) => {
         const componentCandidate = layoutElement as ILayoutComponent;
         let found = false;
 
-        if (validation.field === componentCandidate.id) {
+        if (validation.field && validation.field.match(`${componentCandidate.id}(-[0-9]*)*$`)) {
```

Comment:
Same here... Could this regex expression be a reusable global constant?

Reply:
Yep, pulled it out into a separate function that generates the match.